manifests/server/puppetserver.pp

```diff
@@ -192,6 +193,26 @@
         changes => $jruby_jar_changes,
       }
     }
+
+    $ensure_max_open_files = $max_open_files ? {
+      undef   => 'absent',
+      default => 'present',
+    }
+    if $::service_provider == 'systemd' {
```

Comment:
These days, is it better to use `$facts['service_provider']` rather than the topscope variable?

Reply:
Changed, thanks! :)